src/containers/tasks/TaskForm.jsx

```diff
@@ -58,20 +58,32 @@ function transformMyApiErrors(array) {
   return errors;
 }
 
-function validate(formsById) {
-  return values => {
-    const errors = {};
-    if (values.form && values.form.value) {
-      const theForm = formsById[values.form.value];
-      if (theForm.attributes.metadata.downloadable === false) {
-        errors.form = INACTIVE_XFORM_VALIDATION_MESSAGE;
-      }
-      if (theForm.attributes.metadata.configuration_status !== XFORM_CORRECTLY_CONFIGURED) {
-        errors.form = <MisconfiguredFormMessage />;
+function validate(formsById, fetchForm) {
+  return values =>
+    new Promise(resolve => {
+      const errors = {};
+      if (values.form && values.form.value) {
+        let theForm = formsById[values.form.value];
+
+        if (!theForm) {
+          // Entire FormList is not loaded onto state at times
+          // Especially in update cases as such we try to retrieve the form
+          // if not available in formsById
+          resolve(fetchForm(values.form.value));
+          theForm = formsById[values.form.value];
+
+          if (!theForm) errors.form = 'Could not retrieve form';
+        } else {
+          if (theForm.attributes.metadata.downloadable === false) {
+            errors.form = INACTIVE_XFORM_VALIDATION_MESSAGE;
+          }
+          if (theForm.attributes.metadata.configuration_status !== XFORM_CORRECTLY_CONFIGURED) {
+            errors.form = <MisconfiguredFormMessage />;
+          }
+        }
```

Comment:
This could be a problem, if we don't find the form and we end up loading it using our new `fetchForm` action creator / function, we don't check for this conditions? why?

Reply:
Very true, didn't notice that. Let me change that

Reply:
cool. Also perhaps we should add or update tests for this validate function and all the new code.